### PR TITLE
Potential fix for PR24 bug with resources

### PR DESCRIPTION
**Pull request by TheCodedOne**

### CODE REVIEW

File: resources/src/main/java/org/sandboxpowered/api/resources/ResourceRegistrationService.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.sandboxpowered.api.resources;
+
+import org.sandboxpowered.api.registry.Registrar;
+
+public interface ResourceRegistrationService extends Registrar.Service {
```
Comment (Boundarybreaker) [inline resources/src/main/java/org/sandboxpowered/api/resources/ResourceRegistrationService.java:5]: `requestResource` and `getResource` should probably have their original default impl's.

File: base/src/main/java/org/sandboxpowered/api/registry/Registrar.java [hunk 2]
```diff
@@ -20,11 +19,8 @@
         return register(Identity.of(getSourceAddon().getId(), name), content);
     }
 
-    default Resource requestResource(ResourceRequest request) {
-        return ResourceManager.requestResource(getSourceAddon().getId(), request);
-    }
+    <T extends Service> Optional<T> getRegistrarService(Class<T> tClass);
```
Comment (Boundarybreaker) [inline base/src/main/java/org/sandboxpowered/api/registry/Registrar.java:22]: This is gonna go off the class of the interface, yeah?

File: resources/src/main/java/org/sandboxpowered/api/resources/ResourceRegistrationService.java (hunk 1 above)
Comment (TheCodedOne) [reply resources/src/main/java/org/sandboxpowered/api/resources/ResourceRegistrationService.java:5]: The reason they don't is that they no longer have the getSourceAddon method & I felt like duplicating that or adding a method to get the registrar just bloated the interface

File: base/src/main/java/org/sandboxpowered/api/registry/Registrar.java (hunk 2 above)
Comment (TheCodedOne) [reply base/src/main/java/org/sandboxpowered/api/registry/Registrar.java:22]: Yep

File: resources/src/main/java/org/sandboxpowered/api/resources/ResourceRegistrationService.java (hunk 1 above)
Comment (Boundarybreaker) [reply resources/src/main/java/org/sandboxpowered/api/resources/ResourceRegistrationService.java:5]: well, it is kind of necessary... maybe adding a `void prepareFor(AddonInfo info);` or something would be good?

Comment (TheCodedOne) [reply resources/src/main/java/org/sandboxpowered/api/resources/ResourceRegistrationService.java:5]: Is it, the backend can still do whatever it wants, & the registration services will still be specific to an addon as registrar is. adding prepareFor would just be idiotic to ask users to use

**Pull request merged**